raster/r.horizon/main.c

```diff
@@ -499,6 +499,12 @@ int main(int argc, char *argv[])
         if (nbufferZone == 0.)
             nbufferZone = bufferZone;
 
```

Comment:
I've read up the code and doc to try to understand what the module and bufferzone does, and unless it is checked somewhere else through a call I didn't catch, I think there's a logic flaw which negative values here (and the lines up to the if above). 

In words: the conditions check that if there's any of the arguments that are above 0, then use bufferzone as a default value for the 4 direction-specific buffers.

However I didn't seem to find a place where valid values are checked first. I assume from the formulation of setting default bufferzone to the direction-specific buffers, that only 0 or positive values are valid.

Cases: 
- Negative bufferzone with a positive ebufferzone and other 0 direction-specific bufferzone: the 3 other bufferzones will take a negative value
- positive or negative infinity bufferzone with a positive or positive infinity direction-specific bufferzone, and 3 other 0 direction specific bufferzones: the 3 others will take positive or negative infinity.
- A positive bufferzone, a negative zero ebufferzone, and 3 other positive zero direction-specific bufferzones: the ebufferzone might not have the positive value. 
  - note that I'm not sure about this one. I know that in IEEE754 floating points, the both signed zeros compare true for equality, but what about greater/less than? The sign is also preserved for some mathematical operations. One would need to check a compiled example to see if -0.0 < 0.0. The impact might be later on depending on mathematical operations used.

Reply:
Bufferzone needs to be always positive to include topography outside of the region. It is not checked right now, so perhaps adding `parm.bufferzone->options = "0-";` (and to the rest) would solve this, no?